runner_http.go

```diff
@@ -58,8 +59,11 @@ func (r *runnerHTTP) Run(ctx context.Context, logger *slog.Logger, h Handler) {
 	go func() {
 		select {
 		case <-ctx.Done():
+			shutdownCtx, cancel := context.WithTimeout(context.Background(), 15*time.Second)
```

Comment:
this was a bug in previous impl. We'll give it 15 seconds for now to shutdown, and break the association with the canceled context. This was causing some failures in ci, but it was flaky. Quite strange it was consistently flaking 🤔